go-backend/db/role.go

```diff
@@ -0,0 +1,47 @@
+package db
+
+import (
+	"context"
+
+	logger "github.com/sirupsen/logrus"
+)
+
+// Role - the main struct for a Role object, represents a single role row in the database
+type Role struct {
+	ID   int    `db:"id" json:"id"`
+	Name string `db:"name" json:"name"`
+}
+
+// GetRoleByID - given an ID, retrieve the role object
+func (s *pgStore) GetRoleByID(ctx context.Context, id int) (role Role, err error) {
+	err = s.db.Get(&role, `SELECT * FROM roles WHERE id=$1 LIMIT 1`, id)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error selecting role from database in GetRoleByID for id ", string(id))
+	}
+	return
+}
+
+func (s *pgStore) GetRoleByName(ctx context.Context, name string) (role Role, err error) {
+	err = s.db.Get(&role, `SELECT * FROM roles WHERE name=$1 LIMIT 1`, name)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error selecting role from database in GetRoleByName for name ", name)
+	}
+	return
+}
+
+/*// GetUserByID - Given the database ID for that user, look them up.
+func (s *pgStore) GetUserByID(ctx context.Context, id int) (user User, err error) {
```

Comment:
If not needed, lets get rid of this commented code

Reply:
4c80f054d7bd1df67768281673ccb54f9a095116